SharedTests/NSURLExtensionsTests.swift

```diff
@@ -78,6 +78,17 @@ class NSURLExtensionsTests: XCTestCase {
         XCTAssertEqual("co.uk", expected)
     }
 
+    func testBaseDomainWithTrailingDot() {
```

Comment:
I'd suggest adding three more test cases:
- `"https://foo" => "foo"`
- `"https://foo." => "foo"`
- `"https://." => nil`

Reply:
`foo. => .foo`, let me know if I'm missing something

Reply:
That looks weird, but it's a degenerate case anyways, so I don't think it matters.